storage/remote/write_handler_test.go

```diff
@@ -806,6 +806,197 @@ func TestCommitErr_V1Message(t *testing.T) {
 	require.Equal(t, "commit error\n", string(body))
 }
 
+func TestHistogramValidationErrorHandling(t *testing.T) {
+	testCases := []struct {
+		desc     string
+		hist     histogram.Histogram
+		expected string
+	}{
+		{
+			desc: "count mismatch",
```

Comment:
I wonder if we have to test so many histogram cases. Those are tested in model/histogram package, no? We could test only ~2 to ensure we treat those as 400, that's it, no?